docs/proposals/lvmd-within-topolvm-node.md

```diff
@@ -0,0 +1,312 @@
+	if err := lvcontroller.SetupWithManager(mgr); err != nil {
+		setupLog.Error(err, "unable to create controller", "controller", "LogicalVolume")
+		return err
+	}
+	//+kubebuilder:scaffold:builder
+
+	// Add health checker to manager
+	checker := runners.NewChecker(checkFunc(vgclnt, apiReader), 1*time.Minute)
+	if err := mgr.Add(checker); err != nil {
+		return err
+	}
+
+	// Add metrics exporter to manager.
+	// Note that grpc.ClientConn can be shared with multiple stubs/services.
+	// https://github.com/grpc/grpc-go/tree/master/examples/features/multiplex
+	if err := mgr.Add(runners.NewMetricsExporter(vgclnt, client, nodename)); err != nil {
+		return err
+	}
+
+	// Add gRPC server to manager.
+	if err := os.MkdirAll(driver.DeviceDirectory, 0755); err != nil {
+		return err
+	}
+	grpcServer := grpc.NewServer(grpc.UnaryInterceptor(ErrorLoggingInterceptor))
+	csi.RegisterIdentityServer(grpcServer, driver.NewIdentityServer(checker.Ready))
+	nodeServer, err := driver.NewNodeServer(nodename, vgclnt, lvclnt, mgr)
+	if err != nil {
+		return err
+	}
+	csi.RegisterNodeServer(grpcServer, nodeServer)
+	err = mgr.Add(runners.NewGRPCRunner(grpcServer, config.csiSocket, false))
+	if err != nil {
+		return err
+	}
+
+	setupLog.Info("starting manager")
+	if err := mgr.Start(ctrl.SetupSignalHandler()); err != nil {
+		setupLog.Error(err, "problem running manager")
+		return err
+	}
+
+	return nil
+}
+
+//+kubebuilder:rbac:groups=storage.k8s.io,resources=csidrivers,verbs=get;list;watch
+
+func checkFunc(clnt proto.VGServiceClient, r client.Reader) func() error {
+	return func() error {
+		ctx, cancel := context.WithTimeout(context.Background(), 10*time.Second)
+		defer cancel()
+
+		if _, err := clnt.GetFreeBytes(ctx, &proto.GetFreeBytesRequest{DeviceClass: topolvm.DefaultDeviceClassName}); err != nil {
+			return err
+		}
+
+		var drv storagev1.CSIDriver
+		return r.Get(ctx, types.NamespacedName{Name: topolvm.GetPluginName()}, &drv)
+	}
+}
+
+func ErrorLoggingInterceptor(ctx context.Context, req interface{}, info *grpc.UnaryServerInfo, handler grpc.UnaryHandler) (resp interface{}, err error) {
+	resp, err = handler(ctx, req)
+	if err != nil {
+		ctrl.Log.Error(err, "error on grpc call", "method", info.FullMethod)
+	}
+	return resp, err
+}
+
+func loadConfFile(cfgFilePath string) error {
+	b, err := os.ReadFile(cfgFilePath)
+	if err != nil {
+		return err
+	}
+	err = yaml.Unmarshal(b, config.lvmd)
+	if err != nil {
+		return err
+	}
+	log.Info("configuration file loaded: ", map[string]interface{}{
+		"device_classes": config.lvmd.DeviceClasses,
+		"socket_name":    config.lvmd.SocketName,
+		"file_name":      cfgFilePath,
+	})
+	return nil
+}
+
+```
+
+This will allow starting and using lvmd without grpc communication and another container.
+
+## Caveats
+
+- LVMD can no longer be deployed and managed separately from topolvm-node if started together
+- We will need to decide if we want to offer HELM chart configurations for this style of configuration
```

Comment:
IMO, it is preferable if you could update Helm Chart for the run-lvmd-local mode.
If updating the Helm Chart is not very important for you, it is also OK to work on it later in another PR.

@satoru-takeuchi 
I want to hear your opinion.

Reply:
I want to update the Helm Chart too.